src/index/CompressedRelation.cpp

```diff
@@ -314,111 +314,20 @@ IdTable CompressedRelationReader::scan(
-    firstBlockResult = readIncompleteBlock(*beginBlock);
-    totalResultSize += firstBlockResult.value().size();
-    ++beginBlock;
-    cancellationHandle->throwIfCancelled();
-  }
-  if (beginBlock < endBlock) {
-    lastBlockResult = readIncompleteBlock(*(endBlock - 1));
-    totalResultSize += lastBlockResult.value().size();
-    endBlock--;
-    cancellationHandle->throwIfCancelled();
-  }
-
-  // Determine the total size of the result.
-  // First accumulate the complete blocks in the "middle"
-  totalResultSize += std::accumulate(beginBlock, endBlock, 0UL,
-                                     [](const auto& count, const auto& block) {
-                                       return count + block.numRows_;
-                                     });
-  result.resize(totalResultSize);
-  cancellationHandle->throwIfCancelled();
-
-  size_t rowIndexOfNextBlockStart = 0;
-  // Lambda that appends a possibly incomplete block (the first or last block)
-  // to the `result`.
-  auto addIncompleteBlockIfExists =
-      [&rowIndexOfNextBlockStart, &result](
-          const std::optional<DecompressedBlock>& incompleteBlock) mutable {
-        if (!incompleteBlock.has_value()) {
-          return;
-        }
-        AD_CORRECTNESS_CHECK(incompleteBlock->numColumns() ==
-                             result.numColumns());
-        for (auto i : ad_utility::integerRange(result.numColumns())) {
-          std::ranges::copy(
-              incompleteBlock->getColumn(i),
-              result.getColumn(i).data() + rowIndexOfNextBlockStart);
-        }
-        rowIndexOfNextBlockStart += incompleteBlock->numRows();
-      };
-
-  addIncompleteBlockIfExists(firstBlockResult);
-  cancellationHandle->throwIfCancelled();
-
-  // Insert the complete blocks from the middle in parallel
-  if (beginBlock < endBlock) {
-#pragma omp parallel
-#pragma omp single
-    for (; beginBlock < endBlock; ++beginBlock) {
-      const auto& block = *beginBlock;
-
-      // Read the block serially, only read the second column.
-      AD_CORRECTNESS_CHECK(block.offsetsAndCompressedSize_.size() >= 2);
-      CompressedBlock compressedBuffer =
-          readCompressedBlockFromFile(block, columnIndices);
-
-      // A lambda that owns the compressed block decompresses it to the
-      // correct position in the result. It may safely be run in parallel
-      auto decompressLambda = [rowIndexOfNextBlockStart, &block, &result,
-                               compressedBuffer =
-                                   std::move(compressedBuffer)]() mutable {
-        ad_utility::TimeBlockAndLog tbl{"Decompression a block"};
-
-        decompressBlockToExistingIdTable(compressedBuffer, block.numRows_,
-                                         result, rowIndexOfNextBlockStart);
-      };
-
-      // Register an OpenMP task that performs the decompression of this
-      // block in parallel
-#pragma omp task
-      {
-        if (!cancellationHandle->isCancelled()) {
```

Comment:
This is the only usage of `isCancelled()` outside of tests I think. I'd suggest removing the function or replacing it with something like `ASSERT_THROW(throwIfCancelled())` where it makes sense to keep assertions around.